lib/ramble/ramble/pipeline.py

```diff
@@ -248,12 +248,27 @@ def _prepare(self):
         # - At least one expeirment is set up
         # - `--dry-run` is enabled
         found_valid_experiment = False
-        for exp, app_inst, _ in self._experiment_set.filtered_experiments(self.filters):
+        # Record how many non-analyzable experiments are encountered
+        no_analyze_cnt = 0
+        filtered_exps = list(self._experiment_set.filtered_experiments(self.filters))
+        for _, app_inst, _ in filtered_exps:
             if not (app_inst.is_template or app_inst.repeats.is_repeat_base):
                 if app_inst.get_status() != ramble.application.experiment_status.UNKNOWN.name:
                     found_valid_experiment = True
-
-        if not found_valid_experiment and self._experiment_set.num_experiments():
+            else:
+                no_analyze_cnt += 1
+
+        num_total_exps = self._experiment_set.num_experiments()
+        num_filtered_exps = len(filtered_exps)
```

Comment:
I think instead of keeping a local list of the filtered experiments, we could use `self._experiment_set.num_filtered_experiments(...)`

Reply:
Good call, updated.